wicket-core/src/main/java/org/apache/wicket/markup/head/AbstractJavaScriptReferenceHeaderItem.java

```diff
@@ -135,6 +135,16 @@ public boolean equals(Object o)
 	@Override
 	public int hashCode()
 	{
-		return Objects.hash(async, defer, charset);
+		// original code was: return Objects.hash(async, defer, charset);
+		// This returns the same hash code value, but doesn't allocate an Object[3],
+		// which consumes 32 bytes, and doesn't need to autobox the two booleans.
+		// This method is called very often, especially when a containing Map or Set is resized,
+		// and this version will run faster and save a lot of memory.
+		// It is possible that Escape Analysis would elide the creation of the Object[],
+		// but we do not see that in our running applications with the latest JDK.
+		return 31*31*31 +
+				31*31*Boolean.hashCode(async) +
+				31*Boolean.hashCode(defer) +
+				((charset != null) ? charset.hashCode() : 0);
```

Comment:
I find this quite hard to read. Can you please use IntelliJ or Eclipse to generate the code. IntelliJ looks like this:

```java
	@Override
	public int hashCode() 
	{
		int result = super.hashCode();
		result = 31 * result + (async ? 1 : 0);
		result = 31 * result + (defer ? 1 : 0);
		result = 31 * result + (charset != null ? charset.hashCode() : 0);
		return result;
	}
```

Reply:
I wrote it the original way to produce the exact same hashcode value as the original code, and to minimize the math being done.    Are we OK with the hashcode values being different, perhaps resulting in other changes due to hashing?

I will update the CSSHeaderItem classes, and also remove the comments.

Reply:
> Are we OK with the hashcode values being different, perhaps resulting in other changes due to hashing?

Good question. My guess would be that it doesn't matter, since collecting header items and putting them in collections is always done on a single JVM in a single request. Does anyone else see a problem with changing the hash codes?

Reply:
`hash` method from here https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html can also be used :)